app/src/main/java/me/quadphase/qpdex/databaseAccess/PokemonFactory.java

```diff
@@ -1107,4 +1116,47 @@ public boolean isDetailedNationalIDBuiltAndReady(int nationalID){
         return detailedPokemonShortList!=null && detailedPokemonShortList[nationalID]!=null;
     }
 
+    /**
+     * Constructs a sparse matrix of the type effectivenesses.
+     *
+     * This should be done on initiation of the {@link PokemonFactory}
+     *
+     * @return sparse matrix of the types effectiveness against each other
+     */
+    private double[][] buildTypeEffectivenessTable() {
+        int maxTypeID = getMaxTypeID();
+        double[][] typeEffectiveness = new double[maxTypeID][maxTypeID];
```

Comment:
The point of doing it this way was to make the class's 2D array a final variable. What do you think about this?